cluster-autoscaler/simulator/drainability/filter.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drainability
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/utils/drain"
+
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// Status indicates whether a pod can be drained, with an optional error message when not.
+// TODO(x13n): Move values from drain.BlockingPodReason to some typed string.
+type Status struct {
+	// Decided indicates whether a Filter has made a decision. `false` indicates that
+	// the Filter couldn't make a decision and that another one has to
+	// decide.
+	Decided bool
```

Comment:
That's somewhat surprising - I'd expect most Filter to be able to give confident 'blocking', but not confident 'non-blocking'. In most cases all you know is that a pod can be drained as far as your criteria are considered, but how would you know what other checks are there?
I think your semantics actually make a lot of sense - "safe-to-evict: true" is a use-case for it. It's just really unintuitive and the naming doesn't help. "Undecided status" really isn't undecided, it's not blocking, just without overriding every other Filter. I think renaming things could make it more clear.

Of course I don't really have good naming suggestions... Maybe NewDrainableStatus() for current undecided and NewOverridingDrainableStatus()? But that's a mouthful and it doesn't really seem clear either

Reply:
Hmm, that comment will only really make sense once you do the TODO above though, in the current form the naming actually matches reality pretty well. So I guess the question is should we optimize for end state already?

Reply:
Yeah, I tried to model this after ACLs: each rule either matches (and yields ALLOW/DENY decision) or it doesn't match in which case it is up to the next rule to decide. Would renaming `Decided` to `Matched` (and perhaps - renaming `Filter` to `Rule`) address your concern?